assignments/week_3/exercise_3_2/exercise_3_2.md

```diff
@@ -0,0 +1,111 @@
+# Exercise 3.2
+
+Welcome to exercise 3.2. This exercise will test your ability to create and use pointers. This exercise will expect you to know all the information in the Codecademy C++ course through the References and Pointers section. Our own video on Resources, Lifetime, and Ownership is not needed but will help you get a better understanding of pointers and their use.
+
+# Exercise Statement
+
+In this exercise you will finish writing the code needed to create a linked list. A linked list is a data structure that holds data, similar to a vector. A linked list is made up of nodes in a line each which contains two pieces of information:
+1. Whatever data you want to store in the linked list (in this case a string)
+2. A pointer to the next data
+However, the linked list itself does not keep track of the pointer of each piece of data, but instead only keeps track of the first, or head, node. From the head pointer it is possible to iterate through and access the entire list. You will be creating some functions that help use a linked list, this includes functions that remove data and add new data at specific locations and also one to print the whole list out. Here is a graphical representation of a linked list below.
+
+![](https://www.geeksforgeeks.org/wp-content/uploads/gq/2013/03/Linkedlist.png)
+
+# Starting Off
+
+We have given you a linked_list.hpp header file which contains the class definitions of both the Node class, which stores a name as a string and the pointer to the next node, and the LinkedList class. LinkedList will have one private variable call head, which is a pointer to a Node. Its public members are all methods and are add_to_front, add_to_back, add_at_index, remove_from_front, remove_from_back, remove_from_index, and print_names. It will be your task to define these functions in linked_list.cpp. There is also a constructor which has been done for you. In the linked_list.cpp file you will see the constructors for both Node and LinkedList have been defined for you, but the rest of the functions have empty definitions. 
+
+```c++
+#include <string>
+#include <vector>
+
+class Node {
+public:
+    std::string name;
+    Node *next;
+};
+
+class LinkedList {
+private:
+    Node *head;
+public:
+
+    LinkedList();
+
+    void add_to_front(std::string name);
+
+    void add_to_back(std::string name);
+
+    void add_at_index(std::string name, int index);
+
+    void remove_from_front();
+
+    void remove_from_back();
+
+    void remove_at_index(int index);
+
+    void print_names();
+
+};
+```
+
+# Steps
+### add_to_front:
+1. Create a new variable that is a pointer to a Node
+2. Set the name of the name of the new node to the name parameter passed into the function
+3. Set the new node’s next variable to the head pointer of the linked list
+4. Set the head of the linked list to the new node pointer
+### add_to_back:
+1. Create a new variable that is a pointer to a Node
```

Comment:
I think we need to call out the point that they need to allocate memory here, but IDK how to do that. because new variable to me reads like

Node node;

not 

Node* node = new Node();